storage/localstore/gc.go

```diff
@@ -143,6 +145,12 @@ func (db *DB) collectGarbage() (collectedCount uint64, done bool, err error) {
 		metrics.GetOrRegisterCounter(metricName+".writebatch.err", nil).Inc(1)
 		return 0, false, err
 	}
+	for _, a := range addrs {
```

Comment:
I do not think so. If the batch write fails, that would mean that the reference to an already deleted chunk exists in indexes, which would be considered as a consistency error. If the chunk is first removed from the indexes, it will never be requested from the chunk storage again making the storage more resilient. There can be a cleanup function in that case, but that was not in the scope of this PR. It is a nice improvement to be added. Reversed order would require an immediate action and this order a possible delayed action.

Reply:
> If the chunk is first removed from the indexes, it will never be requested from the chunk storage again making the storage more resilient

but it also means that the underlying chunk store can potentially leak chunks